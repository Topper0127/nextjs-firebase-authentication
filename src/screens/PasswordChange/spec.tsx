import PasswordChangePage from '.';

describe('PasswordChangePage', () => {
  const noSession = null;
  const session = 'session';

  it('does not render when not authorized', () => {
    expect(PasswordChangePage.isAuthorized(noSession)).toEqual(false);
  });

  it('renders when authorized', () => {
    expect(PasswordChangePage.isAuthorized(session)).toEqual(true);
  });
});
