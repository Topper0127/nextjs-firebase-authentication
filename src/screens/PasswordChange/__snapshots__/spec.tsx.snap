// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordChangeFormEnhanced renders 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item Item__StyledFormItem-sc-13gzk8a-0 lhvvBo"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-10"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="/password-change_oldPassword"
        onClick={[Function]}
        title="Old Password"
      >
        Old Password
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-14"
      style={Object {}}
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-password ant-input-affix-wrapper"
          >
            <input
              action="click"
              className="ant-input"
              data-__field={
                Object {
                  "name": "oldPassword",
                }
              }
              data-__meta={
                Object {
                  "name": "oldPassword",
                  "originalProps": Object {
                    "action": "click",
                    "inputPrefixCls": "ant-input",
                    "prefixCls": "ant-input-password",
                    "visibilityToggle": true,
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please input your Password!",
                      "required": true,
                    },
                    Object {
                      "message": "Your password is too short.",
                      "min": 6,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please input your Password!",
                          "required": true,
                        },
                        Object {
                          "message": "Your password is too short.",
                          "min": 6,
                        },
                      ],
                      "trigger": Array [
                        "onBlur",
                      ],
                    },
                  ],
                  "validateFirst": true,
                  "validateTrigger": "onBlur",
                  "valuePropName": "value",
                }
              }
              id="/password-change_oldPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              style={null}
              type="password"
              value=""
            />
            <span
              className="ant-input-suffix"
            >
              <i
                aria-label="icon: eye-invisible"
                className="anticon anticon-eye-invisible ant-input-password-icon"
                onClick={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="eye-invisible"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                  />
                  <path
                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                  />
                </svg>
              </i>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item Item__StyledFormItem-sc-13gzk8a-0 lhvvBo"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-10"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="/password-change_newPassword"
        onClick={[Function]}
        title="New Password"
      >
        New Password
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-14"
      style={Object {}}
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-password ant-input-affix-wrapper"
          >
            <input
              action="click"
              className="ant-input"
              data-__field={
                Object {
                  "name": "newPassword",
                }
              }
              data-__meta={
                Object {
                  "name": "newPassword",
                  "originalProps": Object {
                    "action": "click",
                    "inputPrefixCls": "ant-input",
                    "prefixCls": "ant-input-password",
                    "visibilityToggle": true,
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please input your password!",
                      "required": true,
                    },
                    Object {
                      "message": "Your password is too short.",
                      "min": 6,
                    },
                    Object {
                      "validator": [Function],
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please input your password!",
                          "required": true,
                        },
                        Object {
                          "message": "Your password is too short.",
                          "min": 6,
                        },
                        Object {
                          "validator": [Function],
                        },
                      ],
                      "trigger": Array [
                        "onBlur",
                      ],
                    },
                  ],
                  "validateFirst": true,
                  "validateTrigger": "onBlur",
                  "valuePropName": "value",
                }
              }
              id="/password-change_newPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              style={null}
              type="password"
              value=""
            />
            <span
              className="ant-input-suffix"
            >
              <i
                aria-label="icon: eye-invisible"
                className="anticon anticon-eye-invisible ant-input-password-icon"
                onClick={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="eye-invisible"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                  />
                  <path
                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                  />
                </svg>
              </i>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item Item__StyledFormItem-sc-13gzk8a-0 lhvvBo"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-10"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="/password-change_confirmNewPassword"
        onClick={[Function]}
        title="Confirm Password"
      >
        Confirm Password
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-14"
      style={Object {}}
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-password ant-input-affix-wrapper"
          >
            <input
              action="click"
              className="ant-input"
              data-__field={
                Object {
                  "name": "confirmNewPassword",
                }
              }
              data-__meta={
                Object {
                  "name": "confirmNewPassword",
                  "originalProps": Object {
                    "action": "click",
                    "inputPrefixCls": "ant-input",
                    "onBlur": [Function],
                    "prefixCls": "ant-input-password",
                    "visibilityToggle": true,
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please confirm your password!",
                      "required": true,
                    },
                    Object {
                      "message": "Your password is too short.",
                      "min": 6,
                    },
                    Object {
                      "validator": [Function],
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please confirm your password!",
                          "required": true,
                        },
                        Object {
                          "message": "Your password is too short.",
                          "min": 6,
                        },
                        Object {
                          "validator": [Function],
                        },
                      ],
                      "trigger": Array [
                        "onBlur",
                      ],
                    },
                  ],
                  "validateFirst": true,
                  "validateTrigger": "onBlur",
                  "valuePropName": "value",
                }
              }
              id="/password-change_confirmNewPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              style={null}
              type="password"
              value=""
            />
            <span
              className="ant-input-suffix"
            >
              <i
                aria-label="icon: eye-invisible"
                className="anticon anticon-eye-invisible ant-input-password-icon"
                onClick={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="eye-invisible"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                  />
                  <path
                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                  />
                </svg>
              </i>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item Item__StyledFormItem-sc-13gzk8a-0 lhvvBo"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-control-wrapper ant-col-sm-24"
      style={Object {}}
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <button
            className="ant-btn StretchedButton-sc-1skj5p1-0 gqwrSP ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Change Password
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</form>
`;
